File: bootstrap/src/main/java/net/dontdrinkandroot/wicket/bootstrap/css/FontAwesome5IconClass.java
package net.dontdrinkandroot.wicket.bootstrap.css;

import net.dontdrinkandroot.wicket.css.CssClass;

public enum FontAwesome5IconClass implements CssClass
{
    EDIT("fas fa-edit");

    private final String classString;

    FontAwesome5IconClass(String classString)
    {
        this.classString = classString;
    }

    @Override
    public String getClassString()
    {
        return this.classString;
    }

    public FontAwesome5Icon createIcon()
    {
        return new FontAwesome5Icon(this);
    }
}

File: bootstrap/src/main/kotlin/net/dontdrinkandroot/wicket/bootstrap/css/FontAwesome5IconClass.kt
package net.dontdrinkandroot.wicket.bootstrap.css

import net.dontdrinkandroot.wicket.css.CssClass

enum class FontAwesome5IconClass(override val classString: String) : CssClass {
    ANGLE_DOUBLE_LEFT("fa-angle-double-left"),
    ANGLE_LEFT("fa-angle-left"),
    ANGLE_RIGHT("fa-angle-right"),
    ANGLE_DOUBLE_RIGHT("fa-angle-double-right"),
    ARROWS_ALT_H("fa-arrows-alt-h"),
    DOWNLOAD("fa-download"),
    EDIT("fa-edit"),
    ELLIPSIS_V("fa-ellipsis-v"),
    FILE("fa-file"),
    LOCK("fa-lock"),
    PAUSE("fa-pause"),
    PEN("fa-pen"),
    PLUS("fa-plus"),
    PRINT("fa-print"),
    TRASH("fa-trash"),
    UNDO("fa-undo"),
    UPLOAD("fa-upload");

    fun createIcon(
        style: FontAwesome5Icon.Style = FontAwesome5Icon.Style.SOLID,
        fixedWidth: Boolean = false
    ): FontAwesome5Icon =
        FontAwesome5Icon(this, style = style, fixedWidth = fixedWidth)
}